CMakeLists.txt

```diff
@@ -244,16 +258,70 @@ elseif(CMAKE_GENERATOR MATCHES "Visual Studio")
   add_compile_options("/MP")
 endif()
 
-if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+if(MSVC)
+  if(POLICY CMP0091)
```

Comment:
I think the check should be against cmake version instead (3.22 aligns with our current VS requirements, 3.20 is required for CMP0117 )

Reply:
I'm not sure what advantage this has. By saying `if (POLICY CMP0117)` we state exactly what we care about -- the policy existing -- and not some arbitrary CMake version that could mean anything.

Reply:
I said this because I’d prefer 
cmake_policy(VERSION ${CMAKE_VERSION}) 
but if that’s not done then checking against highest policy we manually enable is fine I guess

Reply:
As in, always set all policies supported on whatever the current CMake versions is on? That seems dangerous and very much invites the behavior changing invisibly when you upgrade CMake, which is kinda what policies are designed to avoid. I'd be okay with `cmake_policy(VERSION 3.22)` (which is the version currently used by VS), though, after testing it works with our files anyway.

Reply:
yes - the thing is that for msvc we know/expect cmake to be the one that gets installed by VS installer. So it's not a random version. I think cmake "getting out of the way" and behaving natively as the one that is installed with msvc is more intuitive and requires less code, but explicitly specifying the version is fine, too (we just have to bump it whenever we bump msvc min version).

anyway, how it is now is fine.